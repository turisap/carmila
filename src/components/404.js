/**
 * Created by HP on 29-Nov-17.
 */
import React from 'react';
import {Link} from 'react-router-dom';

const PageNotFound = () => (
    <div>
        404 <Link to="/">Go home</Link>
    </div>
);

export default PageNotFound;