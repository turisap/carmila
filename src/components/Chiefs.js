/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

const Chiefs = () => (
    <div>
        <p>Chiefs page</p>
    </div>
);

export default Chiefs;