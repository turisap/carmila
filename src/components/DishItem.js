/**
 * Created by HP on 10-Dec-17.
 */
import React from 'react';
import {removeItem, removeAllSameItems, incrementItem} from '../actions/orders';
import {connect} from 'react-redux';

export const DishItem = (props) => (
    <div className="dishItem__content">
        <div className="dishItem__cropper">
            <img src={props.path} className="dishItem__img"/>
        </div>
        <h4 className="dishItem__title">{props.title}</h4>
        {props.orderPage ?
            <p><span className="lead">Price:  </span>{props.price * props.amount}</p>
            :
            <p><span className="lead">Price:  </span>{props.price}</p>
        }
        <p className="dishItem__description">{props.description}</p>
        {(props.amount > 0) && <p className="dishItem__amount"><span className="lead">Amount:  </span>{props.amount}</p>}
        {(props.orderPage && props.amount > 0) && (
            <div className="dishItem__buttonHolder">
                <button className="dishitem__removeone btn btn-primary" onClick={() => props.removeItem(props.title)}>-</button>
                <button className="dishitem__removeall btn btn-primary" onClick={() => props.removeAllSameItems(props.title)}>Remove All</button>
                <button className="dishitem__removeone btn btn-primary" onClick={() => props.addItem(props.title)}>+</button>
            </div>
        )}
    </div>
);

const mapDispatchToProps = (dispatch, props) => ({
    removeItem : (title) => dispatch(removeItem(title)),
    removeAllSameItems : (title) => dispatch(removeAllSameItems(title)),
    addItem : (title) => dispatch(incrementItem(title))
});


export default connect(undefined, mapDispatchToProps)(DishItem);