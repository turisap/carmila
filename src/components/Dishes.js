/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

export default class Dishes extends React.Component{
    render() {
        return (
            <h2>Dishes</h2>
        )
    }
}