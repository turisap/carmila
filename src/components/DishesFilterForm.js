/**
 * Created by HP on 10-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import {categories} from  '../data/data';
import {setPizzaFilter, setPastaFilter, setDessertFilter, setTextFilter} from '../actions/filters';


export class DishesFilterForm extends React.Component {
    state = {
        activeLink : 'pizza'
    };
    handleCategoryChange = (e) => {
        const filter = e.target.text.toUpperCase();
        if (filter === 'PIZZA') {
            this.props.setPizzaFilter();
            this.setState({activeLink : 'pizza'});
        } else if (filter === 'PASTA') {
            this.setState({activeLink : 'pasta'});
            this.props.setPastaFilter()
        } else if (filter === 'DESSERTS') {
            this.setState({activeLink : 'desserts'});
            this.props.setDessertFilter();
        }
        this.props.setTextFilter('')
    };

    handleFilterTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    };

    render () {
        return (
            <div className="dishes__filtersContainer">
                <input className="form-control dishes___textDishFilter" placeholder="Type what you wish" type="text" onKeyDown={this.handleFilterTextChange}/>

                <div className="dishes__textFiltersContainer">
                    {categories.map((c,i) => {
                        return (
                            <a key={i} className={c.toLowerCase() === this.state.activeLink ? 'dishes--activeCategory dishes__text-filter' : 'dishes__text-filter'} onClick={this.handleCategoryChange}>{c}</a>
                        )
                    })}
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        setPizzaFilter : () => dispatch(setPizzaFilter()),
        setPastaFilter : () => dispatch(setPastaFilter()),
        setDessertFilter : () => dispatch(setDessertFilter()),
        setTextFilter  : (text) => dispatch(setTextFilter(text))
    }
};


export default connect(undefined, mapDispatchToProps)(DishesFilterForm);