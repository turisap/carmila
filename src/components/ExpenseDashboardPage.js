/**
 * Created by HP on 29-Nov-17.
 */
import React from 'react';
import ExpenseList from '../components/ExpenseList';
import ExpenseListFilters from '../components/ExpenseListFilters';

const ExpenseDashBoardPage = () => (
    <div>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);

export default ExpenseDashBoardPage;