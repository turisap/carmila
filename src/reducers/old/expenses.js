/**
 * Created by HP on 30-Nov-17.
 */
const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ];
            break;
        case 'REMOVE_EXPENSE':
            return state.filter(({id}) => id !== action.id);
            break;
        case 'EDIT_EXPENSE':
            return state.map(expense => {
                if (expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    }
                }
                return expense;
            });
        default:
            return state;
    }
};