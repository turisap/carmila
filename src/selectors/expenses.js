/**
 * Created by HP on 30-Nov-17.
 */
import moment from 'moment';

const visibleExpenses = (expenses, {text, sortBy, startDate, endDate}) => {
    return expenses.filter(exp => {
        const createdAtMoment = moment(exp.createdAt);
        const textMatch = exp.description.toLowerCase().includes(text.toLowerCase());
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch   = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;

        return textMatch && startDateMatch && endDateMatch;
    }).sort((a,b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });

};

export default visibleExpenses;