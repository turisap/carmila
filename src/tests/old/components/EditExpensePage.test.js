/**
 * Created by HP on 05-Dec-17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage }  from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let wrapper, editExpense, removeExpense, history;

beforeEach(() => {
    history = { push : jest.fn() };
    editExpense = jest.fn();
    removeExpense = jest.fn();
    wrapper = shallow(<EditExpensePage
        editExpense={editExpense}
        removeExpense={removeExpense}
        history={history}
        expense={expenses[0]}
    />);
});

test('should render the editExpensePage component properly', () => {
   expect(wrapper).toMatchSnapshot();
});

test('should handle editExpense() form submission', () => {
   wrapper.find('ExpenseForm').prop('onFormSubmit')(expenses[0]);
   expect(history.push).toHaveBeenCalledWith('/');
   expect(editExpense).toHaveBeenCalledWith(expenses[0].id, expenses[0]);
});

test('should handle removeExpense()', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenCalledWith('/');
    expect(removeExpense).toHaveBeenCalledWith(expenses[0].id);
})