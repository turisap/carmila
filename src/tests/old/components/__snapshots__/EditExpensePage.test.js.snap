// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the editExpensePage component properly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 123,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "",
      }
    }
    onFormSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
